### PR TITLE
Only store swagger.json definitions

### PR DESCRIPTION
**Pull request by spadgett**
sessionStorage has limited space. Only store what we're using for the YAML editor.

### CODE REVIEW

**Comment by spadgett** [comment]
@alecmerdler opinion?

File: frontend/public/components/edit-yaml.jsx [hunk 1]
```diff
@@ -73,8 +73,17 @@ export const EditYAML = connect(stateToProps)(
...
       // Retrieve k8s API spec for autocompletion
       if (!window.sessionStorage.getItem(`${window.SERVER_FLAGS.consoleVersion}--swagger.json`)) {
-        coFetchJSON('api/kubernetes/swagger.json')
-          .then(swagger => window.sessionStorage.setItem(`${window.SERVER_FLAGS.consoleVersion}--swagger.json`, JSON.stringify(swagger)));
+        coFetchJSON('api/kubernetes/swagger.json').then(swagger => {
+          // Only store definitions to reduce the document size.
+          const {definitions} = swagger;
+          const json = JSON.stringify({definitions});
+          try {
+            window.sessionStorage.setItem(`${window.SERVER_FLAGS.consoleVersion}--swagger.json`, json);
```
Comment (alecmerdler) [inline frontend/public/components/edit-yaml.jsx]: Need to change the [YAML editor autocomplete code](https://github.com/openshift/console/blob/master/frontend/public/module/k8s/autocomplete.ts#L142) to read from the new format.

**Comment by spadgett** [comment]
@alecmerdler thanks, updated

File: frontend/__tests__/module/k8s/autocomplete.spec.ts [hunk 2]
```diff
@@ -85,22 +85,20 @@ describe('getCompletions', () => {
...
+          minReadySeconds: {
+            description: 'Dummy property',
+            type: 'integer',
+            format: 'int32',
           },
         },
       },
     };
     sessionMock.getLines = jasmine.createSpy('getLinesSpy').and.returnValue(['kind: Deployment', 'apiVersion: apps/v1']);
-    spyOn(window.localStorage, 'getItem').and.returnValue(JSON.stringify(swagger));
```
Comment (spadgett) [inline frontend/__tests__/module/k8s/autocomplete.spec.ts]: I'm concerned this was passing CI

Comment (alecmerdler) [reply frontend/__tests__/module/k8s/autocomplete.spec.ts]: It's a unit test.

Comment (spadgett) [reply frontend/__tests__/module/k8s/autocomplete.spec.ts]: But we're spying on localStorage, not sessionStorage?

Comment (spadgett) [reply frontend/__tests__/module/k8s/autocomplete.spec.ts]: Interestingly, whichever you spy on (`localStorage` or `sessionStorage`), it seems to work. Removing the spy entirely causes the tests to fail. So I guess it's OK? The tests are running and working.

Comment (alecmerdler) [reply frontend/__tests__/module/k8s/autocomplete.spec.ts]: See `__mocks__/localStorage.ts`.

Comment (spadgett) [reply frontend/__tests__/module/k8s/autocomplete.spec.ts]: Ah ha, that explains it. Thanks.

**Comment by spadgett** [comment]
install failed

/retest

**Comment by spadgett** [comment]
Will need https://github.com/openshift/cluster-authentication-operator/pull/84 before console tests pass

**Comment by spadgett** [comment]
olm flake

/retest

**Comment by spadgett** [comment]
s2i flake

/retest

**Pull request merged**